src/github/client/github-app-client.ts

```diff
@@ -31,13 +30,6 @@ export class GitHubAppClient extends GitHubClient {
 		super(logger, baseUrl);
 		this.appToken = AppTokenHolder.createAppJwt(privateKey, appId);
 
-		this.axios = axios.create({
-			baseURL: this.restApiUrl,
-			transitional: {
-				clarifyTimeoutError: true
-			}
-		});
-
```

Comment:
I moved the axios client out of the concrete GitHub clients and into the `GitHubClient` parent class. This was duplicate code and made it harder to configure the clients in a central place (for example to configure a proxy).

Reply:
pretty sure you could move the interceptors with it as well to the superclass one

Reply:
while you're at it, can you also move the `get`, `post`, `delete`, `put` and `graphql` functions to the superclass?  fanx :)

Reply:
I'm happy to do that (and some other cleanup around the GitHub clients that I think should be done), but I'd prefer to do that in a second PR.